File: MusicManagerAPI/src/main/java/com/ldz/music/manager/model/converter/beans/user/music/status/model/to/bo/UserMusicStatusToBO.java
package com.ldz.music.manager.model.converter.beans.user.music.status.model.to.bo;

import com.ldz.music.manager.model.UserMusicStatus;
import com.ldz.converter.container.ConverterContainer;
import com.ldz.converter.container.annot.BeanConverter;
import com.ldz.converter.container.inter.IConverter;
import com.ldz.music.manager.model.bo.MusicTypeBO;
import com.ldz.music.manager.model.bo.UserMusicStatusBO;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.stream.Collectors;

/**
 * Created by ldalzotto on 03/05/2017.
 */
@Component
@BeanConverter(initialBeanClass = UserMusicStatus.class, targetBeanClass = UserMusicStatusBO.class)
public class UserMusicStatusToBO implements IConverter<UserMusicStatus, UserMusicStatusBO> {

    @Autowired
    ConverterContainer converterContainer;

    @Override
    public UserMusicStatusBO apply(UserMusicStatus userMusicStatus) {
        UserMusicStatusBO userMusicStatusBO = new UserMusicStatusBO();

        if (userMusicStatus != null) {
            userMusicStatusBO.setUsername(userMusicStatus.getUsername());

            userMusicStatusBO.setMusicTypeBOs(
                    userMusicStatus.getMusicTypes().stream().map(musicType -> converterContainer.convert(musicType, MusicTypeBO.class))
                            .collect(Collectors.toList())
            );

        }

        return userMusicStatusBO;
    }
}

File: MusicManagerAPI/src/scala/com/ldz/music/manager/model/converter/model/to/bo/UserMusicStatusToBO.scala
package com.ldz.music.manager.model.converter.model.to.bo

import com.ldz.converter.container.ConverterContainer
import com.ldz.converter.container.annot.BeanConverter
import com.ldz.converter.container.inter.IConverter
import com.ldz.music.manager.model.{MusicType, UserMusicStatus}
import com.ldz.music.manager.model.bo.{MusicTypeBO, UserMusicStatusBO}
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

import scala.collection.JavaConverters._

/**
  * Created by Loic on 20/06/2017.
  */
@Component
@BeanConverter(initialBeanClass = classOf[UserMusicStatus], targetBeanClass = classOf[UserMusicStatusBO])
class UserMusicStatusToBO extends IConverter[UserMusicStatus, UserMusicStatusBO]{

  @Autowired
  private val converterContainer: ConverterContainer = null

  override def apply(userMusicStatus: UserMusicStatus): UserMusicStatusBO = {
    userMusicStatus match {
      case `userMusicStatus` if userMusicStatus != null => {
        val musics = userMusicStatus.getMusicTypes.asScala.map(a => converterContainer.convert(a, classOf[MusicTypeBO]))
        UserMusicStatusBO(userMusicStatus.getUsername, musics)
      }
      case _ => null
    }
  }

}
